atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -905,7 +910,31 @@ public Map<Cell, byte[]> get(TableReference tableRef, Set<Cell> cells) {
                         tableRef,
                         cells,
                         uncached -> getInternal(
-                                "get", tableRef, uncached, immediateKeyValueService, immediateTransactionService));
+                                "get",
+                                tableRef,
+                                uncached,
+                                uncached.size(),
+                                immediateKeyValueService,
+                                immediateTransactionService));
+    }
+
+    @Override
+    public Map<Cell, byte[]> getWithExpectedNumberOfCells(
+            TableReference tableRef, Set<Cell> cells, long expectedNumberOfPresentCells) {
+        return getCache().getWithCachedRef(tableRef, cells, cacheLookupResult -> {
+            long cachedCellsWithNonEmptyValue = cacheLookupResult.cacheHits().values().stream()
+                    .filter(value -> value.value().isPresent() && value.value().get() != PtBytes.EMPTY_BYTE_ARRAY)
```

Comment:
Is this the correct way to check for empty values? I'd have felt more comfortable with a utility method that already exists somewhere that we could reuse everywhere we're checking the byte[]